stdlib/typing_extensions.pyi

```diff
@@ -414,6 +414,13 @@ else:
         # https://github.com/python/typeshed/issues/10224 for why we're defining it this way
         def __buffer__(self, flags: int, /) -> memoryview: ...
 
+# It's invalid to use `Protocol` in a `TypeIs` context, but we need to define it for type checking
+# in the case of narrowing a type to a protocol.
+@type_check_only
+class _Protocol(metaclass=abc.ABCMeta):
```

Comment:
Avoid inheriting from `_ProtocolMeta` here since it doesn't really add anything to the resulting type-definition, and the definition in `typing` differs from the implementation in `typing_extensions` causing some issues.